admin/admin.py

```diff
@@ -210,6 +210,9 @@ def on_model_change(self, form, model, is_created):
             # document has just been validated for the first time
             model.validation_time = datetime.datetime.now()
             got_new_file = model.has_file
+        elif model.validation_time is not None and not model.validated:
+            # document is no longer validated => reset validation time
+            model.validation_time = None
```

Comment:
The early document rewards have a 14 day grace period beginning with the validation of the 5th document. Therefore we need the date of the last actual validation.
Sadly, I missed the the dependency on the bar code re-bake. 
The first solution that comes to mind is to save whether a document has a bar code or not.